watertap/unit_models/tests/test_anaerobic_digestor.py

```diff
@@ -163,269 +130,53 @@ def adm(self):
-        assert (
-            abs(
-                value(
-                    adm.fs.unit.inlet.flow_vol[0] * adm.fs.props.dens_mass
-                    - adm.fs.unit.liquid_outlet.flow_vol[0] * adm.fs.props.dens_mass
-                    - adm.fs.unit.vapor_outlet.flow_vol[0] * adm.fs.props_vap.dens_mass
-                )
-            )
-            <= 1e-6
-        )
-
-        assert pytest.approx(-13.5835, abs=1e-3) == value(
-            adm.fs.unit.liquid_phase.enthalpy_transfer[0]
-        )
-        assert (
-            abs(
-                value(
-                    (
-                        adm.fs.unit.inlet.flow_vol[0]
-                        * adm.fs.props.dens_mass
-                        * adm.fs.props.cp_mass
-                        * (
-                            adm.fs.unit.inlet.temperature[0]
-                            - adm.fs.props.temperature_ref
-                        )
-                    )
-                    - (
-                        adm.fs.unit.liquid_outlet.flow_vol[0]
-                        * adm.fs.props.dens_mass
-                        * adm.fs.props.cp_mass
-                        * (
-                            adm.fs.unit.liquid_outlet.temperature[0]
-                            - adm.fs.props.temperature_ref
-                        )
-                    )
-                    + adm.fs.unit.liquid_phase.enthalpy_transfer[0]
-                )
-            )
-            <= 1e-2
-        )
-
-    @pytest.mark.solver
-    @pytest.mark.skipif(solver is None, reason="Solver not available")
-    @pytest.mark.component
-    def test_costing(self, adm):
-        m = adm
-
-        m.fs.costing = WaterTAPCosting()
-
-        m.fs.unit.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
-        m.fs.costing.cost_process()
-        m.fs.costing.add_LCOW(m.fs.unit.liquid_phase.properties_out[0].flow_vol)
-        solver = get_solver()
-        results = solver.solve(m)
-
-        assert_optimal_termination(results)
-
-        # Check solutions
-        assert pytest.approx(2.0 * 1083290.8, rel=1e-5) == value(
-            m.fs.unit.costing.capital_cost
-        )
-        assert pytest.approx(5.2754, rel=1e-5) == value(m.fs.costing.LCOW)
-
-    @pytest.mark.unit
-    def test_get_performance_contents(self, adm):
-        perf_dict = adm.fs.unit._get_performance_contents()
-
-        assert perf_dict == {
-            "vars": {
-                "Volume": adm.fs.unit.volume_AD[0],
-                "Heat Duty": adm.fs.unit.heat_duty[0],
-            }
+        self.unit_model_block = m.fs.unit
+        self.unit_statistics = {
+            "number_variables": 202,
+            "number_total_constraints": 170,
+            "number_unused_variables": 0,
+        }
+        self.unit_solutions = {
+            "liquid_outlet.pressure[0]": 101325,
+            "liquid_outlet.temperature[0]": 308.15,
+            "liquid_outlet.conc_mass_comp[0, S_I]": 0.328772,
```

Comment:
@MarcusHolly @lbianchi-lbl @bknueven, why don't we make this more similar to the interface of the GUI where we just provide the variable itself? In other words, what's the benefit of providing the string and then using `find_component`?

For example:
```
 self.unit_solutions = {
            m.fs.unit.liquid_outlet.pressure[0]: 101325,
            m.fs.unit.liquid_outlet.temperature[0]: 308.15,
```

Reply:
AFAIK not all of Pyomo's objects are hashable, which is a requirement for Python objects to be used as dictionary keys. I suppose a workaround would be to use a list of tuples instead:

```py
self.unit_solutions = [
    (m.fs.unit.liquid_outlet.pressure[0], 101325),
    (m.fs.unit.liquid_outlet.temperature[0], 308.15),
]
```

The only disadvantage I can see is that the list-of-tuples syntax is somewhat busier visually.